tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -220,6 +220,33 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( "[Blocking] void Bar() { TryBaz( input,out int result ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void TaskToVoidReturnTypeInvocation() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return BarAsync(2); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { { Bar(2); return; } }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeNull() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return null; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { return;}", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeDoubleInvocation() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { if( QuuxAsync() ) { return BarAsync(2); } return BarAsync(4); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { if( Quux() ) { { Bar(2); return; } } { Bar(4); return; } }", actual.Value.ToFullString() );
+	}

```

Comment:
Could I bug you to add cases for the other expressions we allow that make sense?
`return foo++;`, `return foo.bar;`, `return x++` etc.?

Reply:
You can use sharplab to double check if things like `return foo++;` are valid syntax